src/libraries/System.Private.CoreLib/src/System/CodeDom/Compiler/IndentedTextWriter.cs

```diff
@@ -254,5 +256,47 @@ public override void WriteLine(uint value)
             _writer.WriteLine(value);
             _tabsPending = true;
         }
+
+        public override async Task WriteLineAsync()
+        {
+            OutputTabs();
+            await _writer.WriteLineAsync().ConfigureAwait(false);
+            _tabsPending = true;
+        }
+
+        public override async Task WriteLineAsync(char value)
+        {
+            OutputTabs();
+            await _writer.WriteLineAsync(value).ConfigureAwait(false);
+            _tabsPending = true;
+        }
+
+        public override async Task WriteLineAsync(string? value)
+        {
+            OutputTabs();
+            await _writer.WriteLineAsync(value).ConfigureAwait(false);
+            _tabsPending = true;
+        }
+
+        public override async Task WriteLineAsync(StringBuilder? value, CancellationToken cancellationToken = default)
+        {
+            OutputTabs();
+            await _writer.WriteLineAsync(value, cancellationToken).ConfigureAwait(false);
+            _tabsPending = true;
+        }
+
+        public override async Task WriteLineAsync(char[] buffer, int index, int count)
+        {
+            OutputTabs();
+            await _writer.WriteLineAsync(buffer, index, count).ConfigureAwait(false);
+            _tabsPending = true;
+        }
```

Comment:
You can simplify this method by calling the other one and avoid repeated code:
```suggestion
        public override async Task WriteLineAsync(char[] buffer, int index, int count) => WriteLineAsync(buffer.AsMemory(index, count));
```